### DISCUSSION TITLE
Setting up benchmark with detailed weights

### DISCUSSION THREAD

**Discussion by alexismanalo** [dcajasn/Riskfolio-Lib#76] [Q&A] [answered]
Is there a way to setup and refer to a custom benchmark with detailed weights? I'm seeing parameters such as:

`kindbench`
`benchweights`

I'm not sure how to proceed to use these parameters and setup one or more benchmarks (each with different compositions and weights).

Thanks!

**Comment by dcajasn** [comment, marked as answer]
Hi @alexismanalo 

These are the steps:
- kindbench==True
- benchweiths==a dataframe with weights like riskfolio results
- allowTO==True if you want to use turnover or desviation vs benchmark weights constraints.
- turnover==x maximum deviation of benchmark per asset
- allowTE==True if you want tracking error constraints.
- TE==y maximum tracking error

Best,
Dany
